src/java/org/apache/cassandra/locator/ReplicaLayout.java

```diff
@@ -119,6 +119,13 @@ public E pending()
         return pending;
     }
 
+    public int blockFor()
+    {
+        return pending == null
```

Comment:
Seems fragile to use pending not being null without more preconditions asserting pending is not null when it needs to be on the write path? Not sure the assertion in ReplicaLayout's constructor is strong enough?

Reply:
It's a very strong requirement that pending is null on the read path.  I agree this is bad, but it's much better than prior art.  The comment over pending's accessor is meant to imply that this is a known inadequacy, and we will follow up later with a fix that introduces a separate class for Read and Write layouts.  There's other things to do there, too, like a LiveXLayout, and we need to be careful about combinatorial explosion, so it will need some time to ponder carefully.  I think the pending == null semantics will suffice for now, though maybe we should add a comment